### ISSUE TITLE
Remove labels and keep just icons on docks, also create a pattern

### ISSUE THREAD

**Issue by dpizetta** [spyder-ide/spyder#9615]
### Issue Report Checklist

* [X] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [X] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice

## Problem Description

![Screenshot Linux 2019-06-19 14 53 06](https://user-images.githubusercontent.com/5084939/59790229-d4440f00-92a5-11e9-91c4-dc259a19c713.png)

### What is the expected output? What do you see instead?

Remove labels in all windows/docks from toolbuttons (but show tooltip) can provide more space. Also, creating a pattern for the window it will be nice. Or occuping then by name (New, Open/Load, Save, SaveAs, Export/Import). Include the order in plugins' guide too. 

**Toolbuttons | Fields | Menu button**

Example:

**Load Save SaveAs |Undo Redo | Zoom | Search Field | Menu button**

## Versions

- Spyder 4.0.0.dev0 (Commit: 1a079fc93) 
- Python 3.6.8 64-bit
- Qt 5.12.3
- PyQt5 5.12.2
- Linux 4.15.0-52-generic
- QdarkStyle Develop (Commit: 18ef745e2674) 

### Dependencies

pygments >=2.0     :  2.4.2 (OK)
qdarkstyle >=2.6.4 :  2.8 (OK)
sphinx >=0.6.6     :  2.0.1 (OK)
psutil >=0.3       :  5.6.2 (OK)
pyls >=0.19.0;<0.25:  0.24.0 (OK)
nbconvert >=4.0    :  5.5.0 (OK)
pandas >=0.13.1    :  None (NOK)
numpy >=1.7        :  1.16.4 (OK)
sympy >=0.7.3      :  None (NOK)
cython >=0.21      :  None (NOK)
qtconsole >=4.5.0  :  4.5.1 (OK)
IPython >=4.0      :  7.5.0 (OK)
matplotlib >=2.0.0 :  None (NOK)
pylint >=0.25      :  2.3.1 (OK)

**Comment by dpizetta** [comment]
There are more: 

![Screenshot Linux 2019-06-19 15 43 40](https://user-images.githubusercontent.com/5084939/59791932-87face00-92a9-11e9-911b-b354f7f51e3e.png)

![Screenshot Linux 2019-06-19 15 43 24](https://user-images.githubusercontent.com/5084939/59791938-892bfb00-92a9-11e9-83ee-dae8b4ae766e.png)

**Comment by ccordoba12** [comment]
@goanpeca, what do you think about this?

**Comment by goanpeca** [comment]
>Also, creating a pattern for the window it will be nice. Or occuping then by name (New, Open/Load, Save, SaveAs, Export/Import). Include the order in plugins' guide too.

I do agree that we need to make all things uniform so users can expect things without having to change their mindset that one plugin does one thing one way vs another one. This is an unfortunate legacy of being an open source project with no clearly defined style guides, so every time a new things was developed (generally by a different developer) we got a slightly different UI.

Now I am more in favor of consistency than what that actual convention we adopt. I am probably partial to not having text since that makes the interface less crowded and cleaner, but this might affect adoption for less experienced users.

@dpizetta maybe you and @jnsebgosselin could help us define some UI guidelines so we ca try to make things more uniform and leave this written down somewhere so new contributors can see what the guidelines are.

I would like to hear from the rest of the team @spyder-ide/core-developers

**Comment by dpizetta** [comment]
I understand. It is not an easy task to keep all people aligned. I've seen some documenting about your plugins, worked with them and also I working on creating interfaces more or less like yours. So, I can help with what you ask for. I have a friend working with that too.  I 've already some ideas in mind, I'll try to write them down.

About the labels with icons, it is nice to think about the beginners, but we can add an option to show/hide them (this is an option on Qt actions, for example).

Tks!

**Comment by goanpeca** [comment]
>About the labels with icons, it is nice to think about the beginners, but we can add an option to show/hide them (this is an option on Qt actions, for example).

Yes, we should probably add this option globally and leave it on by default? Thoughts @ccordoba12 ?

**Comment by ccordoba12** [comment]
I agree with it.
